web_full_stack/js/src/CategoriesPage.js

```diff
@@ -1,53 +1,48 @@
-import React, { Component, Fragment } from 'react';
+import React, { Fragment, useState, useEffect } from 'react';
 import { fetchCategories } from './API';
 
-export default class CategoriesPage extends Component {
-  state = {
-    categories: [],
-  }
-
-  async componentDidMount() {
-    const resp = await fetchCategories();
+export default function CategoriesPage() {
+  const [categories, setCategories] = useState([]);
+  const [query, setQuery] = useState();
 
-    this.setState({
-      categories: resp.categories,
+  useEffect(() => {
+    fetchCategories().then((response) => {
+      setCategories(response.categories)
     });
-  }
+  }, []);
 
-  onChange = (evt) => {
-    this.setState({ query: evt.target.value });
+  function onChange(event) {
```

Comment:
Can we hoist these functions out of the render function instead of recreating them each time? You will have to pass some args

Reply:
merged too soon. I'll put up another change for this.